### ISSUE TITLE
Android 14 Rockchip - Magisk app says not installed after flashing the patched boot.img

### ISSUE THREAD

**Issue by mo123** [topjohnwu/Magisk#8555]
Device: Rockchip RK3588
Android version: 14
Magisk version name: 28
Magisk version code: 28001

On Rockchip Android 14 arm64 if patching the boot.img with Magisk v28 the patched image is the same size as the stock image even though the Magisk app said patched successfully.

If I use the previous Magisk version then the patched boot.img is about 100KB bigger in size but Magisk also doesn't say Magisk is installed even after flashing the patched boot.img.

After flashing the patched Magisk boot.img then the Magisk app in Android 14 says Ramdisk - Yes, Magisk Installed - N/A so Magisk is not installed and can't give root?

On Android 12 on the same device, Magisk root worked successfully.

Magisk v26.4 and Kitsune Magisk v26.4 works on Android 14, some later commit must have broken something.

Please check if Magisk supports boot images from this Android 14 boot image or how to solve it.
https://nch.pl/s/5EEbywKB82osBn3

**Comment by canyie** [comment]
I haven't look carefully into this due to time reason, but I think it would be better if we have more logs. Please follow https://developer.android.com/studio/debug/bug-report to grab a bugreport and attach it here

**Comment by mo123** [comment]
> I haven't look carefully into this due to time reason, but I think it would be better if we have more logs. Please follow https://developer.android.com/studio/debug/bug-report to grab a bugreport and attach it here

Here is the Magisk install log
https://pastebin.com/raw/LfyUtWBK

I see this error in the log for Magisk v28 but not with Magisk v26.4 where it is working
'unknown/unsupported ASN.1 DER tag: 0x00'

A Google search only leads to this
https://gerrit.twrp.me/c/android_bootable_recovery/+/7762

Will post a logcat later.

**Comment by vendelin8** [comment]
I have the same thing on beryllium.

**Comment by ONEKTM** [comment]
@mo123 @vendelin8  After Android13, Ramdisk was split into init_boot.img. Try fixing this file and then "fastboot flash init_boot magisk_patch-xxxx.img"

**Comment by vendelin8** [comment]
@ONEKTM I don't think so, I've just uncompressed lineage-21.0-20241224-nightly-beryllium-signed.zip and it has the following non-empty files in root:
```
429M system.new.dat.br
290M vendor.new.dat.br
210M product.new.dat.br
131M system_ext.new.dat.br
 64M boot.img
 64M recovery.img
407K odm.new.dat.br
8,0K vbmeta.img
4,7K unsparse_super_empty.img
4,6K system.transfer.list
3,2K vendor.transfer.list
2,8K product.transfer.list
2,1K system_ext.transfer.list
 996 dynamic_partitions_op_list
  22 odm.transfer.list
```

**Comment by vendelin8** [comment]
@ONEKTM @mo123 I rechecked installation, and tried `fastboot flash vbmeta --disable-verity --disable-verification vbmeta.img`. And it works for me.

**Comment by vendelin8** [comment]
Actually, worked that time. Since that I replaced linage with microg lineage, and it fails with the same method. Here's my magisk loading log: https://e.pcloud.link/publink/show?code=XZFDUwZjFGGyCpUMS8tPGFpsC2MJfVqj6m7

**Comment by vendelin8** [comment]
I figured out, that if I change anything in system, even a single build.prop property, Magisk says N/A after reboot. From that point, I need to wipe all partitions except data, flash system image, reboot, flash patched boot and vbmeta. But then it doesn't let me run Revolut.

**Comment by Vojtak42** [comment]
> I figured out, that if I change anything in system, even a single build.prop property, Magisk says N/A after reboot. From that point, I need to wipe all partitions except data, flash system image, reboot, flash patched boot and vbmeta. But then it doesn't let me run Revolut.

But Revolut doesn't mind Magisk at least for me.

**Comment by vendelin8** [comment]
> But Revolut doesn't mind Magisk at least for me.

Not Magisk, but custom rom / open bootloader is a problem for them.
